### ISSUE TITLE
Could not load the PostgreSQLDatabaseProvider DbProviderFactory

### ISSUE THREAD

**Issue by yannicklal** [CollaboratingPlatypus/PetaPoco#646]
I'm use to PetaPoco for years BUT, this time, I have to work with PostgreSQL. Despite of hours of tests I'm unable to connect the default database named postgres and get this error message: Could not load the PostgreSQLDatabaseProvider DbProviderFactory. Note that I can connect the database via psql wihtout difficulties.
I'm using PetaPoco.compiled 6.0.480. Here is my code (C# .NET Framework 4.8) :
`string connStr = "Host=127.0.0.1;Username=postgres;Password=pwd;Database=postgres;Port=5432"
Database myMain = new Database(connStr, new PostgreSQLDatabaseProvider());`

What am I missing ?

**Comment by asherber** [comment]
Does your project include the Npgsql NuGet package? PetaPoco assumes that you have the correct data provider available.

**Comment by yannicklal** [comment]
Yes it does runtime version v4.0.30319

**Comment by asherber** [comment]
That's pretty old -- what's the NuGet package version?

**Comment by yannicklal** [comment]
Nuget package version 6.0.5.0

**Comment by asherber** [comment]
Oh, okay, that's current, not sure why the runtime version is different.

I'm not sure what to say -- the basic code you have above works fine for me (using my own Postgres connection string, of course). It seems like PP is just not finding the Npgsql libraries at runtime.

**Comment by yannicklal** [comment]
Thanks for your reply, indeed I'm using PetaPoco in a library. At runtime Petapoco.dll is copied in the bin directory but Npgsql libraries don't. I'm using VS22 how achieve that copy at runtime?

**Comment by asherber** [comment]
Using it in a library shouldn't matter. If your package references are all correct, then things should get copied correctly.

**Comment by yannicklal** [comment]
Yes I'm agree but how can I check if my package references are all correct? Could you give me an example?

**Comment by asherber** [comment]
If you can post the skeleton of your solution here, or point me to your repo, I can take a look at what you have.

**Comment by yannicklal** [comment]
I sent you an invitation thanks for your precious help!

**Comment by asherber** [comment]
The short answer is that using a project reference like this is not the same as consuming a NuGet package. If Kiwy consumed Kiwy.Core as a NuGet package, then all the dependencies would come along for the ride. But adding a project reference from Kiwy to Kiwy.Core is literally just like adding a reference to Kiwy.Core.dll (except that you can debug into the Kiwy.Core methods) — packages that Kiwy.Core depends on do not get copied over to the Kiwy bin folder.

The solution here is that you need to explicitly add to Kiwy all the NuGet packages that Kiwy.Core needs.
